db/model/generatortemplate.js

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * db/model/generatortemplate.js
+ */
+const common = require('../helpers/common');
+const constants = require('../constants');
+const assoc = {};
+
+const authorSchema = {
+  properties: {
+    url: {
+      description: 'Optional Url given by the author',
+      type: 'string',
+      format: 'url',
+    },
+    name: {
+      description: 'Name of the author',
+      type: 'string',
+      maxLength: constants.fieldlen.normalName,
+      required: true,
+    },
+    email: {
+      description: 'Author\'s email address',
+      type: 'string',
+      format: 'email',
+      require: true,
+    },
+  },
+};
+
+const repositorySchema = {
+  properties: {
+    url: {
+      description: 'Url repository',
+      type: 'string',
+    },
+    type: {
+      description: 'Type of the version control system',
+      type: 'string',
+      maxLength: constants.fieldlen.normalName,
+    },
+  },
+};
+
+const connectionSchema = {
+  properties: {
+    method: {
+      description: 'The http method ',
+      enum: ['DELETE', 'GET', 'PATCH', 'POST', 'PUT'],
+    },
+    url: {
+      description: 'Optional Url given by the author',
```

Comment:
'The url to connect to. Specify variables for variable expansion using double curly braces. One of ["url", "toUrl"] is required.'